### PR TITLE
Export-DbaLogin Respect ScriptingOptionsObject.IncludeDatabaseContext

### LINKED ISSUE #6271: [Bug] Export-DbaUser outputs USE <DB> regardless of ScriptingOptions
Apologies for not filling out the form below.  Line 213 of functions/Export-DbaUser.ps1 is:

`$outsql += "USE [" + $db.Name + "]"`

This negates any ScriptingOptions.IncludeDatabaseContext passed to the function via the -ScriptingOptionsObject param. It even negates the default option set on line 194:

`$ScriptingOptionsObject.IncludeDatabaseContext = $false`

I believe this line 213 should be replaced with something like

`if ($ScriptingOptionsObject.IncludeDatabaseContext) {
    $outsql += "USE [" + $db.Name + "]"
}`

### Environmental information

```powershell
# Replace with output of Get-Culture
```

#### SQL Server: 

```sql
/* REPLACE WITH output of @@VERSION */
```

```sql
/* REPLACE WITH output of @@LANGUAGE */
```

#### Host used

- [ ] powershell.exe
- [ ] ISE
- [x] VS Code
- [ ] Other (please specify)

If anything other than powershell.exe was used, please confirm that you can duplicate the issue with powershell.exe

- [ ] Still buggy in powershell.exe

### PR DESCRIPTION
**Pull request by awickham10**
## Type of Change

 - [x] Bug fix (non-breaking change, fixes #6271 )
 - [ ] New feature (non-breaking change, adds functionality, fixes # )
 - [ ] Breaking change (effects multiple commands or functionality, fixes # )
 - [ ] Ran manual Pester test and has passed (`.\tests\manual.pester.ps1)
 - [ ] Adding code coverage to existing functionality
 - [x] Pester test is included
 - [ ] If new file reference added for test, has is been added to github.com/sqlcollaborative/appveyor-lab ?
 - [ ] Nunit test is included
 - [ ] Documentation
 - [ ] Build system
 

### Purpose
 
Export-DbaLogin does not look at the value of IncludeDatabaseContext in the ScriptingOptionsObject before including a `USE [dbname]` statement in the output.

### Approach

Look at what parameters were specified and if ScriptingOptionsObject is used, make sure to respect what was specified.

### Commands to test

See Pester tests.

### CODE REVIEW

File: functions/Export-DbaUser.ps1 [hunk 1]
```diff
@@ -210,7 +210,9 @@ function Export-DbaUser {
                 Write-ProgressHelper -TotalSteps $users.Count -Activity "Exporting from $($db.Name)" -StepNumber ($stepCounter++) -Message "Generating script for user $dbuser"
 
                 #setting database
-                $outsql += "USE [" + $db.Name + "]"
+                if (($PSBoundParameters.ContainsKey('ScriptingOptionsObject') -and $ScriptingOptionsObject.IncludeDatabaseContext) -or -not $PSBoundParameters.ContainsKey('ScriptingOptionsObject')) {
```
Comment (wsmelton) [inline functions/Export-DbaUser.ps1]:
```suggestion
                if ( ((Test-Bound ScriptingOptionsObject) -and $ScriptingOptionsObject.IncludeDatabaseContext ) -or (Test-Bound ScriptingOptionsObject -Not) ) {
```

We use the internal function across most of our commands so would like to keep this standard.

Comment (awickham10) [reply functions/Export-DbaUser.ps1]: Thanks - I thought there was one but couldn't remember the name. I've updated it, but trying to fix my Pester tests first.

**Comment by awickham10** [comment]
I'm not sure this failure is due to my change.

**Comment by potatoqualitee** [comment]
that command has been failing lately. not you, andrew

**Pull request merged**